pkg/types/chain_writer.go

```diff
@@ -8,10 +8,11 @@ import (
 )
 
 type ChainWriter interface {
-	// SubmitSignedTransaction packs and broadcasts a transaction to the underlying chain.
+	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
-	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
+	// - `args` specifies input parameters to the contract call.
```

Comment:
What are the limitations on the types that can be passed as `args`? b/c we cannot literally support `any`
We need to say something precise, just like `ChainReader` does:
https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/types/chain_reader.go#L25-L29

Reply:
If there is sufficient overlap between the expectations here (even just for serialization), then we may want to declare a new type (alias) that we can document just once, and then use in any related methods, without duplicating the same sort of information around.

@nolag WDYT?

Reply:
Yeah agreed. These args are likely always going to be the arguments passed into the tx to trigger some contract function.

One constraint we have, however, is that the types need to be serializable in some way to be passed over the wire.